### PR TITLE
add: pve-datacenter-manager

### PR DESCRIPTION
**Pull request by CrazyWolf13**
> **🛠️ Note:**  
> We are meticulous about merging code into the main branch, so please understand that pull requests not meeting the project's standards may be rejected. It's never personal!  
> 🎮 **Note for game-related scripts:** These have a lower likelihood of being merged.

---

## ✍️ Description
Provide a summary of the changes made and/or reference the issue being addressed.

 Proxmox recently release a pre-alpha version of Proxmox Datacenter Manager:
 - https://www.proxmox.com/en/about/press-releases/proxmox-datacenter-manager-alpha
 - https://forum.proxmox.com/threads/proxmox-datacenter-manager-first-alpha-release.159323/
 - https://pve.proxmox.com/wiki/Proxmox_Datacenter_Manager_Roadmap

---

## 🛠️ Type of Change
Please check the relevant options:  
- [ ] Bug fix (non-breaking change that resolves an issue)  
- [ ] New feature (non-breaking change that adds functionality)  
- [ ] Breaking change (fix or feature that would cause existing functionality to change unexpectedly)  
- [x] New script (a fully functional and thoroughly tested script or set of scripts)  

---

## ✅ Prerequisites
The following steps must be completed for the pull request to be considered:  
- [x] Self-review performed (I have reviewed my code to ensure it follows established patterns and conventions.)  
- [x] Testing performed (I have thoroughly tested my changes and verified expected functionality.)  
- [x] Documentation updated (I have updated any relevant documentation)

---

## 📋 Additional Information (optional)
Provide any extra context or screenshots about the feature or fix here.  

This Script is based of Proxmox Backup Server, and based off the Contibution Guidelines in the draft PR.

### CODE REVIEW

File: install/pdm-install.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+verb_ip6
+catch_errors
+setting_up_container
+network_check
+update_os
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y curl \
+    sudo \
+    mc
```
Comment (michelroegl-brunner) [inline install/pdm-install.sh:19]:
```suggestion
$STD apt-get install -y \
    curl \
    sudo \
    mc 
```

File: install/pdm-install.sh [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+msg_ok "Installed Dependencies"
+
+msg_info "Installing Proxmox Backup Server"
+wget -q https://enterprise.proxmox.com/debian/proxmox-release-bookworm.gpg -O /etc/apt/trusted.gpg.d/proxmox-release-bookworm.gpg
+cat <<EOF >>/etc/apt/sources.list
+deb http://download.proxmox.com/debian/pdm bookworm pdm-test
+EOF
+$STD apt-get update
+$STD apt-get install -y proxmox-datacenter-manager \
+    proxmox-datacenter-manager-ui
```
Comment (michelroegl-brunner) [inline install/pdm-install.sh:29]:
```suggestion
$STD apt-get install -y \
    proxmox-datacenter-manager \
    proxmox-datacenter-manager-ui
```

File: json/pdm.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      }
+    ],
+    "default_credentials": {
+      "username": null,
+      "password": null
+    },
+    "notes": [
+      {
+        "text": "Set a root password if using autologin. This will be the PBS password. `sudo passwd root`",
+        "type": "warning"
```
Comment (michelroegl-brunner) [inline json/pdm.json:37]:
```suggestion
        "type": "info"
```

**Review by michelroegl-brunner — COMMENTED** [review]
Script looks fine besides a few nitpicky things. 

But i´m not sure if we should add Software which is in Alpha?

File: json/pdm.json (hunk 3 above)
Comment (CrazyWolf13) [reply json/pdm.json:37]: I'd have used info here as well, but on pbs it was also a warning, but will correct :)

**Comment by CrazyWolf13** [comment]
@michelroegl-brunner 
Thanks for the review, I've fixed the things mentioned.

About whether to merge or not, I think a lot of users would love to test a "new" Proxmox Project, as it's a long and often requested thing, but a warning about the alpha state may be a good idea, I'll add one.

File: install/pdm-install.sh [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+msg_info "Installing Dependencies"
+$STD apt-get install -y \
+    curl \
+    sudo \
+    mc 
+msg_ok "Installed Dependencies"
+
+msg_info "Installing Proxmox Backup Server"
+wget -q https://enterprise.proxmox.com/debian/proxmox-release-bookworm.gpg -O /etc/apt/trusted.gpg.d/proxmox-release-bookworm.gpg
```
[the remark this answers is not in the document]
Comment (CrazyWolf13) [reply install/pdm-install.sh:24]: Not sure if I fully understand this right?
Is there any example ?

Something like this?
```bash
echo "-----BEGIN PGP PUBLIC KEY BLOCK-----
<KEY>
-----END PGP PUBLIC KEY BLOCK-----" | gpg --dearmor -o /etc/apt/trusted.gpg.d/proxmox-release-bookworm.gpg
```

[the remark this answers is not in the document]
Comment (michelroegl-brunner) [reply install/pdm-install.sh:24]: more like that.
```bash
curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_20.x nodistro main" >/etc/apt/sources.list.d/nodesource.list
```

[the remark this answers is not in the document]
Comment (CrazyWolf13) [reply install/pdm-install.sh]: Thanks @michelroegl-brunner Yup at that part it makes more sense, was a bit confused as the actual echo was not part of the review, but fixed it in the new commit.

File: json/proxmox-datacenter-manager.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+        }
+      }
+    ],
+    "default_credentials": {
+      "username": null,
+      "password": null
+    },
+    "notes": [
+      {
+        "text": "Set a root password if using autologin. This will be the PBS password. `sudo passwd root`",
```
Comment (mh166) [inline json/proxmox-datacenter-manager.json:36]: "PBS" should probably be changed. ;)

File: ct/proxmox-datacenter-manager.sh [hunk 6]
```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+source <(curl -s https://raw.githubusercontent.com/community-scripts/ProxmoxVE/raw/main/misc/build.func)
+# Copyright (c) 2021-2024 community-scripts ORG
+# Author: CrazyWolf13
+# License: MIT | https://github.com/community-scripts/ProxmoxVE/raw/main/LICENSE
+# Source: Proxmox Server Solution GmbH
+
+# App Default Values
+APP="proxmox-datacenter-manager"
+var_tags="backup"
```
Comment (michelroegl-brunner) [inline ct/proxmox-datacenter-manager.sh:10]:
```suggestion
var_tags="server,datacenter"
```
It is not only backup. But mine are just a suggestion, maybe you can come up with somethig better.

File: ct/proxmox-datacenter-manager.sh [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+# Core
+variables
+color
+catch_errors
+
+function update_script() {
+    header_info
+    check_container_storage
+    check_container_resources
+    if [[ ! -d /var ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
```
Comment (michelroegl-brunner) [inline ct/proxmox-datacenter-manager.sh:31]: Can you check where it actually gets installed and put the path here.

**Review by michelroegl-brunner — CHANGES_REQUESTED** [review]
Just to little things left, the i would approve it.

File: ct/proxmox-datacenter-manager.sh (hunk 7 above)
Comment (CrazyWolf13) [reply ct/proxmox-datacenter-manager.sh]: Sure, I tought as this is just an apt package, and tteck did it like this in pbs, I tought I'll stick to it, however sure, just added a path to the proxmox-datacenter-manager admin binary.

**Review by michelroegl-brunner — DISMISSED** [review]

**Pull request merged**